File: A. Soldier and Bananas.cpp
#include <iostream>
using namespace std;

//Solver : Nahin the Coder
//Date   : 06-03-2020
//Time   : 10:43

int main() {
    int k,n,w;
    int counter = 0;
    cin>>k>>n>>w;
    for(int i = 1; i<=w; i++)
    {
        counter=counter+(i*k);
    }
    if(counter<=n)
    {
        cout<<0<<endl;
    }
    else
    {
        cout<<counter-n<<endl;
    }

    return 0;
}

File: Solved by Python/A. Soldier and Bananas.py
# Nahin The Coder
# Date : 20 - 07 - 2021

k, n, w = map(int, input().split())

start = int(1)
counter = int(0)
while start <= w:
    counter = counter + int(start * k)
    start = start + 1

if counter <= n:
    print(0)
else:
    print(counter - n)
